my-code/src/components/LikeButton/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Container, Button } from './styles';
+import iconHeartWhite from './assets/icon-heart-white.svg';
+import iconHeartGrey from './assets/icon-heart-grey.svg';
+import iconHeartFull from './assets/icon-heart-full.svg';
+
+export default function LikeButton({ imdbID, showText, alwaysVisible }) {
+  const [liked, setLiked] = useState(false);
+  const dispatch = useDispatch();
+
+  // Redux state: get the favourites list.
+  const likes = useSelector(state => state.likesReducer);
+
+  // Based on the favourites list, set the liked state.
+  useEffect(() => {
+    setLiked(likes.indexOf(imdbID) >= 0);
+  }, []);
```

Comment:
Actually you have two dependencies on this effect: `likes` and `imdbID`.
Check your `eslint` rules and if your editor code is validating this kind of rule.

Reply:
I had used an empty array to execute the effect only once as recommended by the useEffect documentation. The value of imdbID will not change over time and the component just have to know if the movie is already liked when it renders the first time to show an empty or a filled heart icon.

After adding the dependencies the effect is executed 10 times (1 time per movie on screen) after each like button is clicked on the search results page while using an empty array this doesn't happens.

Am I thinking the wrong way?

I've added the dependencies anyway since I adjusted my ESLINT and now it is auto-fixing this on save and it didn't broke the app.